api/routes/upload_refactored.py

```
@@ -0,0 +1,112 @@
+        filename = request.filename or request.fileName
+        file_size = request.file_size or request.fileSize
+        
+        return upload_service.init_upload(
+            filename=filename,
+            file_size=file_size,
+            chunk_size=request.chunk_size,
+            is_admin=False
+        )
+    except Exception as e:
+        logger.error(f"Failed to init upload: {e}")
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)